crates/mqtt-format/src/v5/packets/suback.rs

```diff
@@ -32,6 +32,33 @@ crate::v5::reason_code::make_combined_reason_code! {
     }
 }
 
+impl SubackReasonCode {
+    /// Return Some(_) if self is a "Granted" reason code, otherwise None
+    #[must_use]
+    pub fn granted(&self) -> Option<Self> {
+        match self {
+            SubackReasonCode::GrantedQoS0
+            | SubackReasonCode::GrantedQoS1
+            | SubackReasonCode::GrantedQoS2 => Some(*self),
+            _ => None,
+        }
+    }
+
+    /// Check whether a certain quality of service was granted
+    #[must_use]
+    pub fn qos_granted(&self, qos: impl AsRef<crate::v5::qos::QualityOfService>) -> bool {
+        use crate::v5::qos::QualityOfService::*;
+
+        matches!(
+            (self, qos.as_ref()),
+            (SubackReasonCode::GrantedQoS0, AtMostOnce)
+                | (SubackReasonCode::GrantedQoS1, AtMostOnce)
+                | (SubackReasonCode::GrantedQoS1, AtLeastOnce)
+                | (SubackReasonCode::GrantedQoS2, _)
+        )
+    }
+}
```

Comment:
```suggestion
    pub fn granted(&self) -> Option<crate::v5::qos::QualityOfService> {
        match self {
            SubackReasonCode::GrantedQoS0 => Some(crate::v5::qos::QualityOfService::AtMostOnce),
            SubackReasonCode::GrantedQoS1 => Some(crate::v5::qos::QualityOfService::AtLeastOnce),
            SubackReasonCode::GrantedQoS2 => Some(crate::v5::qos::QualityOfService::ExactlyOnce),
            _ => None,
        }
    }
}
```

I was thinking like this!